components/builder-depot/src/handlers/integrations.rs

```diff
@@ -55,22 +54,12 @@ pub fn validate_params(
             res.insert(p.to_string(), params.find(p).unwrap().to_string());
         }
     }
-    // Check that we have origin access
-    {
-        let session_id = {
-            req.extensions.get::<Authenticated>().unwrap().get_id()
-        };
-        if !check_origin_access(req, session_id, &res["origin"])
-            .map_err(|_| status::InternalServerError)?
-        {
-            debug!(
-                "Failed origin access check, session: {}, origin: {}",
-                session_id,
-                &res["origin"]
-            );
-            return Err(status::Forbidden);
-        }
+
+    if !check_origin_access(req, &res["origin"]).unwrap_or(false) {
```

Comment:
We could even take the origin check out of validate_params now, and plumb it outside in the integration calls.  validate_params and get_param could potentially also be merged. Not a requirement for this PR tho.

Reply:
Yeah, good idea.  I think we could simplify the way we're checking for the presence of a param and returning a 400 if it's not there, too.  But, I didn't want to do too much refactoring in 1 PR.